src/mlpack/methods/ann/loss_functions/KLDivergence.hpp

```diff
@@ -0,0 +1,112 @@
+/**
+ * @file KLDivergence.hpp
+ * @author Dakshit Agrawal
+ *
+ * Definition of the KL Divergence error function.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_METHODS_ANN_LOSS_FUNCTION_KLDIVERGENCE_HPP
+#define MLPACK_METHODS_ANN_LOSS_FUNCTION_KLDIVERGENCE_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * The KL divergence error function measures the network's
```

Comment:
I think this one could be improved, e.g. The Kullback–Leibler divergence is often used  for continuous distributions (direct regression). Perhaps adding a reference  (paper) is useful as well.

Reply:
Should this reference be like:
@code
  @article{
    title={On Information and Sufficiency},
    author={S. Kullback, R.A. Leibler},
    url = {https://projecteuclid.org/euclid.aoms/1177729694},
    journal = {The Annals of Mathematical Statistics},
    year={1951}
  }
 @endcode